monitor/data_loader.py

```diff
@@ -0,0 +1,264 @@
+        Args:
+            index (:obj:`str`): The index to which we want to load data.
+            field (:obj:`str`): The field of the data to be loaded.
+            value (:obj:`str`): The value of the data to be loaded.
+    
+        """
+
+        body = {
+            "doc.{}".format(field): value,
+            "doc.time": time.time() 
+        }
+        
+        resp = self.es.index(index, body)
+
+    @staticmethod 
+    def gen_data(index, data):
+        """ 
+        Formats logs so it can be sent to Elasticsearch in bulk.
+    
+        Args:
+            log_data (:obj:`list`): A list of logs in dict form.
+    
+        Yields:
+            :obj:`dict`: A dict conforming to the required format for sending data to elasticsearch in bulk.
+        """
+    
+        for log in data:
+            yield {
+                "_index": index,
+                "doc": log
+            }
+    
+    def index_data_bulk(self, index, data):
+        """ 
+        Indexes logs in elasticsearch so they can be searched.
+    
+        Args:
+            index (:obj:`str`): The index to which we want to load data.
+            data (:obj:`list`): A list of data in dict form.
+    
+        Returns:
+            response (:obj:`tuple`): The first value of the tuple equals the number of the logs data was entered successfully. If there are errors the second value in the tuple includes the errors.
+    
+        Raises: 
+            elasticsearch.helpers.errors.BulkIndexError: Returned by Elasticsearch if indexing log data fails.
+            
+        """
+    
+        response = helpers.bulk(self.es, self.gen_data(index, data))
+    
+        # The response is a tuple of two items: 1) The number of items successfully indexed. 2) Any errors returned.
+        if (response[0] <= 0):
+            logger.error("None of the logs were indexed. Log data might be in the wrong form.") 
+
+        return response
+
+    def get_num_appointments(self):
+        """ 
+        Gets number of appointments the tower is storing in the watcher and responder, so we can load this data into Elasticsearch.
+    
+        Returns:
+            :obj:`list`: A list where the 0th element describes # of watcher appointments and the 1st element describes # of responder appointments.
+        """
+
+        teos_url = "http://{}:{}".format(self.api_host, self.api_port)
+
+        resp = teos_cli.get_all_appointments(teos_url)
+
+        response = json.loads(resp)
+
+        watcher_appts = len(response.get("watcher_appointments"))
+        responder_appts = len(response.get("responder_trackers"))
+
+        return [watcher_appts, responder_appts]
+
+    def delete_index(self, index):
+        """ 
+        Deletes the chosen index of Elasticsearch.
+    
+        Args:
+            index (:obj:`str`): The ES index to delete.
+        """
+
+        results = self.index_client.delete(index)
+
+    # For testing purposes...
+    def search_logs(self, field, keyword, index):
```

Comment:
This does not seem to be used anymore. If the only purpose is testing, maybe it could be part of a local temp file or the testing suite.